azure/iac/Redis.rego

```diff
@@ -0,0 +1,95 @@
+package rule
+
+# https://docs.microsoft.com/en-us/azure/templates/Microsoft.Cache/redis
+
+# PR-AZR-0131-ARM
+
+default serverRole = null
+
+azure_attribute_absence ["serverRole"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.cache/redis"
+    not resource.properties.enableNonSslPort
+}
+
+azure_issue ["serverRole"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.cache/redis"
+    resource.properties.enableNonSslPort != false
+}
+
+serverRole {
+    azure_attribute_absence["serverRole"]
+}
+
+
+serverRole {
+    lower(input.resources[_].type) == "microsoft.cache/redis"
+    not azure_issue["serverRole"]
+}
+
+
+serverRole = false {
+    azure_issue["serverRole"]
+}
+
+serverRole_err = "Azure Redis Cache enableNonSslPort is not false currently on the resource" {
+    azure_issue["serverRole"]
+}
+
+
+serverRole_metadata := {
+    "Policy Code": "PR-AZR-0131-ARM",
+    "Type": "IaC",
+    "Product": "AZR",
+    "Language": "ARM template",
+    "Policy Title": "Ensure that the Redis Cache accepts only SSL connections",
+    "Policy Description": "It is recommended that Redis Cache should allow only SSL connections. Note: some Redis tools (like redis-cli) do not support SSL. When using such tools plain connection ports should be enabled.",
+    "Resource Type": "Microsoft.Cache/redis",
+    "Policy Help URL": "",
+    "Resource Help URL": "https://docs.microsoft.com/en-us/azure/templates/Microsoft.Cache/redis"
+}
+
+
+
+
+# https://docs.microsoft.com/en-us/azure/templates/microsoft.cache/redis/linkedservers
+
+# PR-AZR-0132-ARM
+
+default serverRole = null
+
+
+azure_issue ["serverRole"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.cache/redis/linkedservers"
+    lower(resource.properties.serverRole) != "secondary"
```

Comment:
we should check the property existence.
Though this property is required and customer somehow may forgot to add this property.

Reply:
if customer somehow forgot to add the objects in arm template which they required, he/she can't deploy that

Reply:
@shahinshirvani  Yes they can't deploy and we are doing the validation before deployment. is n't it better if we can notify customer before deployment, that a required property is missing from your template?